lumberjack.go

```diff
@@ -244,8 +259,8 @@ func (l *Logger) openNew() error {
 // backupName creates a new filename from the given name, inserting a timestamp
 // between the filename and the extension, using the local time if requested
 // (otherwise UTC).
-func backupName(name string, local bool) string {
-	dir := filepath.Dir(name)
+func (l *Logger) backupName(name string, local bool) string {
+	dir := l.backupDir()
```

Comment:
pass `filepath.Dir(name)` as parameter to `backupDir` to keep backward compatibility (have this as first default, and second default is `l.dir()`)

Reply:
Since `name` is always `l.filename()`, and since `l.dir()` is based on it - we already keep backward compatibility.